pkg/kubeconfig/kubeconfig_writer.go

```diff
@@ -0,0 +1,113 @@
+package kubeconfig
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/aws/eks-anywhere/pkg/clients/kubernetes"
+	"github.com/aws/eks-anywhere/pkg/clusterapi"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+)
+
+// ClientFactory builds Kubernetes clients.
+type ClientFactory interface {
+	// BuildClientFromKubeconfig builds a Kubernetes client from a kubeconfig file.
+	BuildClientFromKubeconfig(kubeconfigPath string) (kubernetes.Client, error)
+}
+
+// GenericKubeconfigWriter reads the kubeconfig secret on a cluster and copies the contents to a writer.
+type GenericKubeconfigWriter struct {
```

Comment:
This name is not very informative. Generic doesn't provide any new info and it's a term that in software is used for more than one thing, so it's ambiguous.

What about `ClusterAPIKubeconfigSecretWriter`?